warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/JexlNodes.java

```diff
@@ -199,4 +202,39 @@ public static ASTNumberLiteral literal(Number n) {
         l.literal = n;
         return l;
     }
+    
+    /**
+     * Remove childToRemove from parent, updating all references on both the parent and childToRemove to be consistent
+     * 
+     * @param parent
+     *            the parent to remove from
+     * @param childToRemove
+     *            the child to remove from parent
+     * @return true if childToRemove was successfully found and removed from parent, false otherwise
+     */
+    public static boolean removeFromParent(JexlNode parent, JexlNode childToRemove) {
+        boolean found = false;
+        // at most as many children as currently exist
+        List<JexlNode> children = new ArrayList<>(parent.jjtGetNumChildren());
+        JexlNode[] nodeArray = new JexlNode[0];
+        
+        for (int i = 0; i < parent.jjtGetNumChildren(); i++) {
+            JexlNode child = parent.jjtGetChild(i);
+            if (!child.equals(childToRemove)) {
```

Comment:
.equals isn't implemented for JexlNode or it's descendants.  This could probably just be a simple '=='.  Also, if child is null, this will throw an exception.